contracts/CompoundAssetProxy.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity >=0.5.16;
+
+import "./interfaces/IERC20.sol";
+import "./WrappedPosition.sol";
+import "./interfaces/CTokenInterfaces.sol";
+import "./interfaces/ComptrollerInterface.sol";
+import "./libraries/Authorizable.sol";
+
+/// @author Element Finance
+/// @title Compound Asset Proxy
+contract CompoundAssetProxy is WrappedPosition, Authorizable {
+    uint8 public immutable underlyingDecimals;
+    CErc20Interface public immutable ctoken;
+    ComptrollerInterface public immutable comptroller;
+
+    /// @notice Constructs this contract and stores needed data
+    /// @param _ctoken The compound ctoken contract
+    /// @param _comptroller The compound comptroller
+    /// @param _token The underlying token
+    /// @param _name The name of the token created
+    /// @param _symbol The symbol of the token created
+    constructor(
+        address _ctoken,
+        address _comptroller,
+        IERC20 _token,
```

Comment:
We should take an owner address and set the owner to it so that it's not always the deployer and we don't have to make a followup tx